compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1646,8 +1646,7 @@ private void populateIntersectionTypeReferencedFunctions(DataInputStream inputSt
                 BInvokableType attachedFuncType = (BInvokableType) readTypeFromCp();
                 Name funcName = names.fromString(Symbols.getAttachedFuncSymbolName(
                         objectSymbol.name.value, attachedFuncName));
-                Name funcOrigName = names.fromString(Symbols.getAttachedFuncSymbolName(
-                        objectSymbol.originalName.value, attachedFuncName));
+                Name funcOrigName = names.fromString(attachedFuncName);
```

Comment:
There's a slight issue here in the BIR. It seems we are not writing the original name to the BIR as we do for regular function definitions. For regular functions defs: https://github.com/ballerina-platform/ballerina-lang/blob/a8f28183cee016fab227886b95aa3047315b5eb2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRBinaryWriter.java#L208

For attached functions, not writing the original name: https://github.com/ballerina-platform/ballerina-lang/blob/a8f28183cee016fab227886b95aa3047315b5eb2/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java#L493